[Add Check]: Slashes in names and missing/placeholder descriptions (#139)
NeurodataWithoutBorders/nwbinspector: docs/best_practices/general.rst

the class name (e.g. "ElectricalSeries_1" and "ElectricalSeries_2"). If one of the instances is an extra of less
importance, name that one something different (e.g. "ElectricalSeries" and "ElectricalSeries_extra_electrode").

.. _best_practice_name_slashes:

Do Not Use Slashes
~~~~~~~~~~~~~~~~~~

'/' is not allowed in Group or Dataset names, as this can confuse h5py and lead to the creation of an additional group.
Instead of including a forward slash in the name, please use "Over" like in DfOverF.

Check function: :py:meth:`~nwbinspector.checks.general.check_name_slashes`

Descriptions
------------

.. _best_practice_description:

Metadata and Descriptions
~~~~~~~~~~~~~~~~~~~~~~~~~

> Names are not for storing meta-data. If you need to place other data of the same neurodata_type, you will need to
> choose another name. Keep in mind that meta-data should not be stored solely in the name of objects. It is OK to name

Reviewer: ```suggestion
Names are not for storing meta-data, so meta-data should not be stored solely in the name of objects. It is OK to name
```

an object something like “ElectricalSeries_large_array” however the name alone is not sufficient documentation. In this
case, the source of the signal will be clear from the device of the rows from the linked electrodes table region, and
you should also include any important distinguishing information in the description field of the object. Make an effort
to make meta-data as explicit as possible. Good names help users but do not help applications parse your file.

> As such, it is not recommended to use blank or default 'placeholder' descriptions.

Reviewer: It seems like this section is touching on 3 separate issues:
* Storing metadata in names
* Using default names
* Good descriptions

Author: This section was improved (or attempting to be improved) overall in PR #130 

Specific section changes can be found here: https://github.com/NeurodataWithoutBorders/nwbinspector/pull/130/files#diff-9b30fc25c0f88f5894cf6c4667d4ca96ea980382040a527d808c41b5ec55b60fR32-R51

I think this specific PR only added a little specific content about placeholders and added reference links

the class name (e.g. "ElectricalSeries_1" and "ElectricalSeries_2"). If one of the instances is an extra of less
importance, name that one something different (e.g. "ElectricalSeries" and "ElectricalSeries_extra_electrode").

.. _best_practice_name_slashes:

No Not Use Slashes
~~~~~~~~~~~~~~~~~~

> ’/’ is not allowed in names. When creating a custom name, using the forward slash (/) is not allowed, as this can

Reviewer: ```suggestion
’/’ is not allowed in Group or Dataset names, as this can
```

Author: Do references to Groups and Datasets still make sense on the MatNWB side?

Reviewer: Yes, because those are [HDMF schema language ](https://hdmf-schema-language.readthedocs.io/en/latest/) concepts (and by extension, NWB schema language concepts).
